Builds/Test.py

```diff
@@ -227,45 +324,163 @@ def run_build(args=None):
                 print(*lines, sep='')
             sys.exit(1)
 
+def get_cmake_dir(cmake_dir):
+    return os.path.join('build' , 'cmake' , cmake_dir)
+
+def run_cmake(directory, cmake_dir, args):
+    print('Generating build in', directory, 'with', *args or ('default options',))
+    old_dir = os.getcwd()
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+    os.chdir(directory)
+    if IS_WINDOWS and not any(arg.startswith("-G") for arg in args) and not os.path.exists("CMakeCache.txt"):
+        if '--ninja' in args:
+            args += ( '-GNinja', )
+        else:
+            args += ( '-GVisual Studio 14 2015 Win64', )
+    args += ( '-Dtarget=' + cmake_dir, os.path.join('..', '..', '..'), )
+    resultcode, lines = shell('cmake', args)
+
+    if resultcode:
+        print('Generating FAILED:')
+        if not ARGS.verbose:
+            print(*lines, sep='')
+        sys.exit(1)
 
-def main():
-    if ARGS.all:
-        to_build = ALL_BUILDS
-    else:
-        to_build = [tuple(ARGS.scons_args)]
+    os.chdir(old_dir)
+
+def run_cmake_build(directory, target, config, args):
+    print('Building', target, config, 'in', directory, 'with', *args or ('default options',))
+    build_args=('--build', directory)
+    if target:
+      build_args += ('--target', target)
+    if config:
+      build_args += ('--config', config)
+    if args:
+        build_args += ('--',)
+        build_args += tuple(args)
```

Comment:
Cmake build mode does not abstract verbosity away, so we'd have to determine the correct flag for every build type (`msbuild` uses something like `-verbosity:d`, `make` uses `-d`, `ninja` uses `-v`, etc), and make sure we had the correct build type (which is sometimes implied, and not explicitly on the command line like `-GNinja`). I'm not sure it's worth it...